include/deepworks/shape.hpp

```diff
@@ -1,9 +1,30 @@
 #pragma once
 
 #include <vector>
+#include <ostream>
 
 namespace deepworks {
 
 using Shape = std::vector<int>;
 
+inline std::ostream& operator<<(std::ostream& os, const Shape& shape) {
```

Comment:
Why this implementation is needed?

Reply:
Because the `std::vector` doesn't have this implementation ? Not sure that understand your question

Reply:
I got your point there were two implementations, fixed